### PR TITLE
Use libmesh_error_msg_if() where appropriate

### PR DESCRIPTION
**Pull request by jwpeterson**
Refs #2660.

### CODE REVIEW

File: src/apps/compare.C [hunk 1]
```diff
@@ -152,10 +148,10 @@ void process_cmd_line(int argc,
            */
         case 'r':
           {
-            if ((!names.empty()) && (left_name_set))
-              names.push_back(optarg);
-            else
-              libmesh_error_msg("ERROR: Mesh file name and left file name must precede right file name!");
+            libmesh_error_msg_if(names.empty() || !left_name_set,
```
Comment (roystgnr) [inline src/apps/compare.C:151]: > [I don't trust myself](https://github.com/libMesh/libmesh/pull/2660#issuecomment-664539859) to DeMorgan-ize complicated compound logical expressions

I thought that line was a good explanation for your decision; now I fear it was also foreshadowing...

File: src/systems/fem_system.C [hunk 2]
```diff
@@ -1392,8 +1392,7 @@ void FEMSystem::mesh_position_get()
 {
   // This function makes no sense unless we've already picked out some
   // variable(s) to reflect mesh position coordinates
-  if (!_mesh_sys)
-    libmesh_error_msg("_mesh_sys was nullptr!");
```
Comment (roystgnr) [inline src/systems/fem_system.C:1396]: In hindsight I probably should have made this an assert.

File: src/apps/compare.C (hunk 1 above)
Comment (jwpeterson) [reply src/apps/compare.C:151]: Not gonna lie, I have the [wikipedia page](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) open in one of my tabs right now.

**Comment by jwpeterson** [comment]
Thanks for reviewing!

File: src/reduced_basis/rb_eim_evaluation.C [hunk 3]
```diff
@@ -292,10 +280,8 @@ Number RBEIMEvaluation::get_eim_basis_function_value(unsigned int basis_function
                                                      unsigned int comp,
                                                      unsigned int qp) const
 {
-  if (basis_function_index >= _local_eim_basis_functions.size())
-  {
-    libmesh_error_msg("Invalid basis function index: " + std::to_string(basis_function_index));
-  }
+  libmesh_error_msg_if(basis_function_index >= _local_eim_basis_functions.size(),
+                       "Invalid basis function index: " + basis_function_index);
```
Comment (roystgnr) [inline src/reduced_basis/rb_eim_evaluation.C:284]: You dropped the std::to_string() here.

Perhaps more alarmingly, this made it through CI!?!  Even the --enable-paranoid-warnings --enable-werror runs!?!

It looks like ```-Wstring-plus-int``` is a clang warning but not yet a gcc warning, and we only use --enable-werror on gcc builds.  Right now we're in "we'll take what we can get" mode on CI, but after that's straightened out we ought to just turn on --enable-paranoid-warnings --enable-werror on every CI build.

Comment (jwpeterson) [reply src/reduced_basis/rb_eim_evaluation.C:284]: Ha, I meant to change the` +` into a `<<`, I'll push that now...

**Pull request merged**